index.js

```diff
@@ -11,28 +11,163 @@
+          player1.shipCount = player1.shipCount - 1;
+          //update ship count in the page
+          document.getElementById("ships_player1").innerHTML =
+            player1.shipCount;
+          if (player1.shipCount === 0) {
+            alert(player2.name + " you WIN!");
+            //we restart the game
+            newGame();
+          }
+        } //with this propertie you can change the background color of the clicked cell. try comment the line bellow and uncomment this line. Do not forget to save this file and refresh the borwser to see the changes
+        //switch players
+        player1.isMyTurn = true;
+        player2.isMyTurn = false;
+        document.getElementById("turn_player").innerHTML = player1.name;
       });
+      li.appendChild(cell); //adding each cell into the row number x
+    }
+    board_Player1.appendChild(li); //adding each row into the board
+  }
+};
 
+//Create board player 2
+const createBoardPlayer2 = () => {
+  const board_Player2 = document.getElementById("board_player2");
+  for (var x = 0; x < 4; x++) {
+    const li = document.createElement("li"); // creating childs for the list (board), in this case represent a row number 'x' of the board
+    for (var y = 0; y < 4; y++) {
+      const cell = document.createElement("div");
+      cell.className = "square"; // adding css properties to make it looks like a square
+      cell.textContent = `${x},${y}`; // saves the coordinates as a string value 'x,y'
+      cell.value = 0; //state of the cell
+      if (player2.gameBoard[y][x] === 1) {
+        cell.setAttribute("has-ship", "1");
+      } else {
+        cell.setAttribute("has-ship", "0");
+      }
+      //this function adds the click event to each cell
+      cell.addEventListener("click", (e) => {
+        let cell = e.target; // get the element clicked
+        console.log(cell.textContent); //display the coordinates in the console
+        if (cell.attributes["has-ship"].value === "0") {
+          cell.style.visibility = "hidden"; // this  means that the contents of the element will be invisible, but the element stays in its original position and size / try it clicking on any of the black cells (in your browser) and see whats happens
+        } else {
+          cell.style.background = "purple";
+          player2.shipCount = player2.shipCount - 1;
+          //update ship count in the page
+          document.getElementById("ships_player2").innerHTML =
+            player2.shipCount;
+          if (player2.shipCount === 0) {
+            alert(player1.name + "  you WIN!");
+            //restart the game
+            newGame();
+          }
+        }
+        //switch players
+        player2.isMyTurn = true;
+        player1.isMyTurn = false;
+        document.getElementById("turn_player").innerHTML = player2.name;
+      });
       li.appendChild(cell); //adding each cell into the row number x
     }
+    board_Player2.appendChild(li); //adding each row into the board
+  }
+};
```

Comment:
`createBoardPlayer1` and `createBoardPlayer2` do almost the same thing. We can optimize this code having only one function like `createBoardPlayer` and pass player1 or player2 as arguments (we would have to adapt the code inside the function a little bit depending of the player we are receiving). Also, if that big function can be split into several smaller functions, it would make the code more modular.